clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -932,20 +942,31 @@ template <typename Derived> class SyclKernelFieldHandler {
   // class/field graph. Int Headers use this to calculate offset, most others
   // don't have a need for these.
 
-  virtual void enterStruct(const CXXRecordDecl *, FieldDecl *) {}
-  virtual void leaveStruct(const CXXRecordDecl *, FieldDecl *) {}
-  virtual void enterStruct(const CXXRecordDecl *, const CXXBaseSpecifier &) {}
-  virtual void leaveStruct(const CXXRecordDecl *, const CXXBaseSpecifier &) {}
+  virtual bool enterStruct(const CXXRecordDecl *, FieldDecl *) { return true; }
```

Comment:
Are these changes now required?

Reply:
Yes, all functions called using the visitor model must return bool.